src/main/java/projectI/Parser.java

```diff
@@ -0,0 +1,61 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import org.javatuples.Pair;
+
+public class Parser {
+    final String[] keywords = {"var", "is", "type", "record", "end",
+           "array", "if", "then", "else", "routine",
+           "length", "size"};
+
+    final String[] operators = {"and", "or", "xor",
+        "not", "<", "<=", ">", "=", "/=", "*", "/", "%",
+        "+", "-", ".."};
+
+    final String[] symbols = {"[", "]", "."};
+
+    public enum TokenKind {
+        Keyword,
+        Identifier,
+        Separator,
+        Operator,
+        Type
+    }
+
+    public static String readString(String str) throws IOException {
```

Comment:
I suggest to indicate that argument is path to file, having just `str` is ambiguous.

Reply:
This file is a draft of another implementation. Is should be removed in future